swift/ql/lib/codeql/swift/generated/IpaTypes.qll

```diff
@@ -0,0 +1,442 @@
+private import codeql.swift.elements.IpaConstructors
+
+cached
+module Cached {
+  cached
+  newtype TElement =
+    TFile(@file id) or
+    TLocation(@location id) or
+    TAccessorDecl(@accessor_decl id) or
+    TAssociatedTypeDecl(@associated_type_decl id) or
+    TClassDecl(@class_decl id) or
+    TConcreteFuncDecl(@concrete_func_decl id) or
+    TConcreteVarDecl(@concrete_var_decl id) or
+    TConstructorDecl(@constructor_decl id) or
+    TDestructorDecl(@destructor_decl id) or
+    TEnumCaseDecl(@enum_case_decl id) or
+    TEnumDecl(@enum_decl id) or
+    TEnumElementDecl(@enum_element_decl id) or
+    TExtensionDecl(@extension_decl id) or
+    TGenericTypeParamDecl(@generic_type_param_decl id) or
+    TIfConfigClause(@if_config_clause id) or
+    TIfConfigDecl(@if_config_decl id) or
+    TImportDecl(@import_decl id) or
+    TInfixOperatorDecl(@infix_operator_decl id) or
+    TMissingMemberDecl(@missing_member_decl id) or
+    TModuleDecl(@module_decl id) or
+    TOpaqueTypeDecl(@opaque_type_decl id) or
+    TParamDecl(@param_decl id) or
+    TPatternBindingDecl(@pattern_binding_decl id) or
+    TPostfixOperatorDecl(@postfix_operator_decl id) or
+    TPoundDiagnosticDecl(@pound_diagnostic_decl id) or
+    TPrecedenceGroupDecl(@precedence_group_decl id) or
+    TPrefixOperatorDecl(@prefix_operator_decl id) or
+    TProtocolDecl(@protocol_decl id) or
+    TStructDecl(@struct_decl id) or
+    TSubscriptDecl(@subscript_decl id) or
+    TTopLevelCodeDecl(@top_level_code_decl id) or
+    TTypeAliasDecl(@type_alias_decl id) or
+    TAnyHashableErasureExpr(@any_hashable_erasure_expr id) or
+    TAppliedPropertyWrapperExpr(@applied_property_wrapper_expr id) or
+    TArchetypeToSuperExpr(@archetype_to_super_expr id) or
+    TArgument(@argument id) or
+    TArrayExpr(@array_expr id) or
+    TArrayToPointerExpr(@array_to_pointer_expr id) or
+    TArrowExpr(@arrow_expr id) or
+    TAssignExpr(@assign_expr id) or
+    TAutoClosureExpr(@auto_closure_expr id) or
+    TAwaitExpr(@await_expr id) or
+    TBinaryExpr(@binary_expr id) or
+    TBindOptionalExpr(@bind_optional_expr id) or
+    TBooleanLiteralExpr(@boolean_literal_expr id) or
+    TBridgeFromObjCExpr(@bridge_from_obj_c_expr id) or
+    TBridgeToObjCExpr(@bridge_to_obj_c_expr id) or
+    TCallExpr(@call_expr id) { not constructMethodCallExpr(id) } or
```

Comment:
Thinking about this a bit more, I don't think we need to automatically generate these `not x1(...) and not x2(...) and ... not xn(...)` branches because:
1. This forces us to map all db entities to _some_ IPA branch (which we might not want). For instance, I think we want to hide away the [SelfApplyExpr](https://github.com/apple/swift/blob/main/include/swift/AST/Expr.h#L4703) entity since we want to provide a different API for obtaining qualifiers.
2. If we do need a catch-all IPA branch like this we can just create it in the `schema.yml` file with some `_ipa: ... on:` fields, I think

Reply:
I don't completely get it, I guess we can chat about it later :slightly_smiling_face: This avoids the same db instances appearing in separate synthesized classes, wouldn't that be a problem?

Reply:
I'm trying out generalising constructors to all types, with useful defaults (`none()` for actual IPA type, and `not constructX` disjunctions for DB types, with `any()` if no ipa type is built from it), but want to keep an eye if it impacts test run time.